includes/rest-api.php

```diff
@@ -318,17 +320,46 @@ function pmpro_rest_api_get_has_membership_access($request) {
 		 */
 		function pmpro_rest_api_change_membership_level( $request ) {
 			$params = $request->get_params();
-			$user_id = isset( $params['user_id'] ) ? intval( $params['user_id'] ) : null;
-			$level_id = isset( $params['level_id'] ) ? intval( $params['level_id'] ) : null;
+			$user_id = isset( $params['user_id'] ) ? (int) $params['user_id'] : null;
+			$level_id = isset( $params['level_id'] ) ? (int) $params['level_id'] : null;
 			$email = isset( $params['email'] ) ? sanitize_email( $params['email'] ) : null;
+			$create_user = isset( $params['create_user'] ) ? (bool) $params['create_user'] : false;
 			$response_type = isset( $params['response_type'] ) ? sanitize_text_field( $params['response_type'] ) : null;
 
 			if ( empty( $user_id ) ) {
+
 				// see if they sent an email
 				if ( ! empty( $email ) ) {
+
 					$user = get_user_by_email( $email );
-					$user_id = $user->ID;
+					
+					// Assume the user doesn't already exist.
+					if ( $create_user && ! $user ) {
+
+						$user_id = wp_create_user( $email, wp_generate_password(), $email );
+						
+						/**
+						 * Send the default WordPress emails when creating a WordPress user, using the /change_membership_level route.
+						 *
+						 * @since TBD
+						 *
+						 * @param boolean $bool Should default WordPress emails be sent to both the admins and users.
+						 */
+						if ( apply_filters( 'pmpro_api_create_user_notifications', false ) ) {
+							wp_send_new_user_notifications( $user_id, 'both' );
+						}
+
+					} else {
+						$user_id = $user->ID;
+
+						// Make sure the user isn't an admin. If they are throw an error.
+						if ( user_can( $user_id, 'manage_options' ) ) {
```

Comment:
We might want to add a filter to allow someone to bypass this

Reply:
Added new filter in, the wording can catch you out of the filter but tried to be as self explanatory as I could.

Reply:
It makes sense if you say "pmpro_api_change_membership_level_disallow_admin_change", Yes and No in terms of reading it in a sentence.

Reply:
Defaults to Yes (true) so admins won't be affected or allowed to change their level.